onnxruntime/core/providers/cpu/tensor/col2im.cc

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <cassert>
+#include "core/providers/cpu/tensor/col2im.h"
+
+#include "core/framework/element_type_lists.h"
+#include "core/framework/TensorSeq.h"
+#include "core/providers/common.h"
+#include "core/framework/copy.h"
+#include "core/common/safeint.h"
+#include "core/providers/op_kernel_type_control.h"
+
+namespace onnxruntime {
+
+#define REGISTER_KERNEL_TYPED(T)                                                            \
+  ONNX_CPU_OPERATOR_TYPED_KERNEL(                                                           \
+      Col2Im,                                                                               \
+      1,                                                                                    \
+      T,                                                                                    \
+      KernelDefBuilder().TypeConstraint("T", DataTypeImpl::AllTensorTypes()), \
+      Col2Im<T>);
+
+REGISTER_KERNEL_TYPED(float)
+
+template <typename T>
+Status Col2Im<T>::Compute(OpKernelContext* context) const {
+  const auto* col_input = context->Input<Tensor>(0);
+  const auto* image_shape = context->Input<Tensor>(1);
+  const auto* kernel_shape = context->Input<Tensor>(2);
+  std::cout << "Status Col2Im<T>::Compute(OpKernelContext* context)" << std::endl;
+
+  const T* col_input_data = col_input->template Data<T>();
+  TensorShape col_input_shape = col_input->Shape();
+  int64_t col_input_C = col_input_shape[1];
+  const auto col_input_N = col_input_shape[0];
+
+  int64_t image_shape_size = 1;
+  int64_t kernel_shape_size = 1;
+  int64_t kernel_shape_rank = 0;
+  for (auto i=0; i < image_shape->Shape().Size(); ++i) {
+    ++kernel_shape_rank;
+    image_shape_size *=  image_shape->Data<int64_t>()[i];
+    kernel_shape_size *=  kernel_shape->Data<int64_t>()[i];
+    // col_input_C computed as => (C*n-ary-prod{kernel_shape}) / n-ary-prod{kernel_shape}
+    col_input_C /= kernel_shape->Data<int64_t>()[i];
+  }
+  const int64_t col_input_offset = col_input_C * image_shape_size;
```

Comment:
Don't understand this. Why multiply by `image_shape_size` ? Also nit: `offset` doesn't seem a good name, this is more like `size` or `stride`?

Reply:
How about replacing the RHS by `col_input_shape.SizeFromDimension(1)` ?

Reply:
And renaming the variable to `col_input_stride` ?

Reply:
Renamed and used `col_input_shape.SizeFromDimension(1)`